packages/vite/src/node/plugins/resolve.ts

```diff
@@ -923,29 +923,28 @@ export function resolvePackageEntry(
     return cached
   }
   try {
-    let entryPoint: string | undefined | void
+    let entryPoints: string[] = []
 
-    // resolve exports field with highest priority
-    // using https://github.com/lukeed/resolve.exports
+    // the exports field takes highest priority as described in
+    // https://nodejs.org/api/packages.html#package-entry-points
     if (data.exports) {
-      entryPoint = resolveExports(data, '.', options, targetWeb)
-    }
-
-    // if exports resolved to .mjs, still resolve other fields.
-    // This is because .mjs files can technically import .cjs files which would
-    // make them invalid for pure ESM environments - so if other module/browser
-    // fields are present, prioritize those instead.
-    if (
-      targetWeb &&
-      options.browserField &&
-      (!entryPoint || entryPoint.endsWith('.mjs'))
-    ) {
+      entryPoints = resolveExports(
+        data,
+        '.',
+        options,
+        getInlineConditions(options.conditions, targetWeb)
+      )
+      if (!entryPoints.length) {
+        packageEntryFailure(id)
```

Comment:
This comes from #8484, and it makes resolution more strict when an `exports` field is found. If the imported module isn't specified in `exports` map, avoid checking other fields (e.g. `main`).